src/common/ipc/update-available/index.ts

```diff
@@ -0,0 +1,48 @@
+import { UpdateInfo } from "electron-updater";
+
+export const UpdateAvailableChannel = "update-available";
+export const AutoUpdateLogPrefix = "[UPDATE-CHECKER]";
+
+/**
+ * [<back-channel>, <update-info>]
+ */
+export type UpdateAvailableFromMain = [string, UpdateInfo];
+
+export function areArgsUpdateAvailableFromMain(args: unknown[]): args is UpdateAvailableFromMain {
```

Comment:
What exactly these typing mean? `args is UpdateAvailableFromMain` and at the same time `args: unknown[]`? Is it possible to determine argument types instead of `unknown`?

Reply:
The is what Typescipt calls a "user defined type-guard" (https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards).

Namely, it is a function that is used to validate that the inputs are a certain type and thus narrow them (if the function returns `true`).